simple-cpp-stream/src/core/MediaController.cpp

```
@@ -0,0 +1,92 @@
+
+StreamContext MediaController::startStream(int client_fd, HttpRequest& req, std::string fileName, int epoll_fd) {
+    StreamContext ctx;
+    if (fileName.find("..") != std::string::npos) { sendError(client_fd, 403, "Forbidden"); return ctx; }
+
+    std::string file_path = VIDEO_DIR + fileName;
+    int file_fd = open(file_path.c_str(), O_RDONLY);
+    if (file_fd < 0) { sendError(client_fd, 404, "File Not Found"); return ctx; }
+
+    struct stat file_stat;
+    if (fstat(file_fd, &file_stat) < 0) {
+        close(file_fd)
```

Cppcheck (2.19.0)

[error] There is an unknown macro here somewhere. Configuration is required. If close is a macro then please configure it.

(unknownMacro)